app/views/layouts/_head.html.erb

```diff
@@ -36,6 +36,17 @@
   <%= include_account_css %>
   <script>
     window.bz_custom_css_url = <%= custom_css_url.to_json.html_safe %>;
+    try {
+      if(window.location.pathname != '/' && window.location.pathname.indexOf('/login') != 0) {
+        var message = "";
+        message += 'last_url=' + encodeURIComponent(window.location.href) + "&";
+        message += 'last_url_title=' + encodeURIComponent(document.title);
+        var request = new XMLHttpRequest();
+        request.open("POST", "/bz/last_user_url", true);
```

Comment:
On Thu, Jul 02, 2015 at 08:51:25AM -0700, Brian Sadler wrote:
> Yikes.  This worries me a little from a performance perspective.

Canvas is already dog slow, every load has like 15 http requests
as it is.... one more async one doesn't make much of a difference
(I measured this too in the firefox network thing)

> Is there no MVC way to accomplish this instead of using javascript to post to the route?

The beauty of a javascript thing is it ensures it only stores
a page that the user actually sees; it won't be triggered
on an api request, resource load, or transition path, and
can be triggered by a partial page reload.

I considered making the view itself update the database which
is the next best thing for one location that covers the right
things but not api requests, but that is just kinda ugly code
and still doesn't cover the partial page case (since the view
isn't generated there... but the user does see the change).